tests/test_api.rs

```diff
@@ -4255,3 +4258,31 @@ fn value_serializer_not_implemented() {
     "Uncaught Error: Deno serializer: get_shared_array_buffer_id not implemented"
   );
 }
+
+#[test]
+fn clear_kept_objects() {
+  let _setup_guard = setup();
+
+  let isolate = &mut v8::Isolate::new(Default::default());
+  isolate.set_microtasks_policy(v8::MicrotasksPolicy::Explicit);
+
+  let scope = &mut v8::HandleScope::new(isolate);
+  let context = v8::Context::new(scope);
+  let scope = &mut v8::ContextScope::new(scope, context);
+
+  let step1 = r#"
+    var weakrefs = [];
+    for (let i = 0; i < 424242; i++) weakrefs.push(new WeakRef({ i }));
```

Comment:
424242 might seem a little excessive but it needs to be big enough to get the GC to come online - and it completes in milliseconds on my PC.